File: playwright/javascript/locators/AccountLocators.js
module.exports = {
  // Navigation and Search
  appLauncherButton: 'button[aria-label="App Launcher"]',
  appLauncherSearchInput: 'input.slds-input[placeholder="Search apps and items..."]',
  accountsNavLink: 'nav a[data-label="Accounts"]',
  globalSearchInput: 'input[placeholder="Search..."]',

  // Account Record Details
  accountLinkByName: (accountName) => `a[title="${accountName}"]`,
  tcvAmountDisplayValue: 'lightning-output-field[field-name="TCV_Amount__c"] lightning-formatted-number',
  
  // Edit Actions
  editButton: 'button[name="EditButton"]',
  tcvAmountEditInput: 'lightning-input-field[field-name="TCV_Amount__c"] input',
  saveButton: 'button[name="SaveEdit"]',

  // Error Messages
  validationErrorMessage: 'div.forceVisualMessageQueue ul li',
  
  // Existing locator for record ID (from context)
  recordId: '.forceOutputLookup a'
};

File: selenium/java/locators/AccountLocators.java
package locators;

import org.openqa.selenium.By;

public class AccountLocators {
    // Global search input in Salesforce header
    public static final By GLOBAL_SEARCH_INPUT = By.cssSelector("input[placeholder='Search Salesforce']");

    // Account search result link by name in the search results
    public static By ACCOUNT_SEARCH_RESULT_LINK_BY_NAME(String accountName) {
        return By.xpath("//a[@title='" + accountName + "']");
    }

    // Account Record Page - View Mode
    // General header containing the account title (from elem_4yem9e)
    public static final By ACCOUNT_PAGE_TITLE = By.xpath("//h1[contains(@class, 'slds-page-header__title')]");
    // TCV Amount field on the detail page in view mode
    public static final By TCV_AMOUNT_FIELD_VIEW_MODE = By.cssSelector("lightning-output-field[field-name='TCV_Amount__c'] lightning-formatted-text");
    // Edit button on the Account record page highlight panel (from 'edit')
    public static final By EDIT_BUTTON = By.xpath("//button[@name='Edit']");

    // Account Record Page - Edit Modal
    // TCV Amount input field in the edit modal (targeting the actual input element)
    public static final By TCV_AMOUNT_INPUT_EDIT_MODE = By.cssSelector("lightning-input-field[field-name='TCV_Amount__c'] input[type='text']");
    // Save button in the edit modal (from 'saveEdit')
    public static final By SAVE_BUTTON = By.cssSelector("button[name='SaveEdit']");

    private AccountLocators() {}
}